app/src/componentLibrary/modal/modalLayout/modalLayout.jsx

```diff
@@ -42,14 +44,28 @@ export const ModalLayout = ({
   allowCloseOutside,
 }) => {
   const [isShown, setShown] = useState(false);
+  const [modalHeight, setModalHeight] = useState(0);
   const modalRef = useRef();
 
+  const windowSize = useWindowResize();
+  const windowHeight = windowSize.height;
+  const modalMaxHeight = windowHeight * MODAL_MAX_RATIO;
+  const modalPadding = (windowHeight - modalHeight) / 2;
+  const contentMaxHeight = modalMaxHeight - MODAL_HEADER_AND_FOOTER_HEIGHT;
+
   const onKeydown = (e) => {
     if (e.keyCode === ESC_KEYCODE) {
       setShown(false);
     }
   };
 
+  useEffect(() => {
+    if (modalRef && modalRef.current) {
+      const { clientHeight } = modalRef.current;
+      setModalHeight(clientHeight);
+    }
+  }, [title, headerNode, children, footerNode, modalHeight, modalRef, windowSize]);
```

Comment:
I think that `title, headerNode,footerNode, modalHeight,modalRef` is redundant dependencies here.